core/views.py

```diff
@@ -1,22 +1,72 @@
-from rest_framework import permissions
-from rest_framework import generics
-from rest_framework import status
-from .serializers import HackathonSerializer
-from .models import Hackathon
-from .permissions import HackathonPermissions
-
-class HackathonCreateView(generics.CreateAPIView):
-    '''
-    API used to create hackathon objects. Can only be accessed by the Super User
-    '''
+from rest_framework import generics, status, permissions, exceptions
+from rest_framework.response import Response
+from django.utils import timezone
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+from .models import Hackathon, Team
+from .serializers import HackathonSerializer, TeamSerializer, TeamCreateSerializer
+from .permissions import HackathonPermissions, AllowCompleteProfile
+
+class HackathonTeamView(generics.ListCreateAPIView):
+    permission_classes = [permissions.IsAuthenticated, AllowCompleteProfile]
+
+    def get_serializer_class(self):
+        if self.request.method == 'GET':
+            return TeamSerializer
+        else:
+            return TeamCreateSerializer
+
+    def get_queryset(self, **kwargs):
+        if getattr(self, 'swagger_fake_view', False):
+            return None
+        try:
+            return Hackathon.objects.get(id=self.kwargs['pk'])
```

Comment:
why return the hackathon?

Reply:
To check if the hackathon with that id exists or not, although instead of ```return``` it should just be
```hackathon = Hackathon.objects.get(id=self.kwargs['pk'])```